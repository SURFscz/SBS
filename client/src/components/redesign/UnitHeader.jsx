import React from "react";
import "./UnitHeader.scss";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import Logo from "./Logo";
import {isEmpty, splitListSemantically, stopEvent} from "../../utils/Utils";
import PropTypes from "prop-types";
import Button from "../Button";
import {ButtonType, Chip, MenuButton} from "@surfnet/sds";
import {Link} from "react-router-dom";
import I18n from "../../locale/I18n";
import {MoreLessText} from "../MoreLessText";

class UnitHeader extends React.Component {

    constructor(props, context) {
        super(props, context);
        this.state = {
            showDropDown: false
        };
    }

    performAction = action => e => {
        stopEvent(e);
        !action.disabled && action.perform();
    }

    otherOptions = (chevronActions, firstTime, auditLogPath, history, queryParam) => {
        return (
            <ul className={"other-options"}>
                {chevronActions.map((action, index) => <li key={index} onClick={this.performAction(action)}>
                    <a href={`/${action.name}`}>{action.name}</a>
                </li>)}
                {(history && auditLogPath) &&
                    <li onClick={() => this.props.history.push(`/audit-logs/${auditLogPath}?${queryParam}`)}>
                        <Link to={`/audit-logs/${auditLogPath}?${queryParam}`}>
                            {I18n.t("home.history")}
                        </Link>
                    </li>}
                {firstTime &&
                    <li onClick={this.performAction({perform: firstTime})}>
                        <a href={"/" + I18n.t("home.firstTime")}>
                            {I18n.t("home.firstTime")}
                        </a>
                    </li>}
            </ul>
        )
    }

    render() {
        const {
            obj, history, auditLogPath, name, breadcrumbName, svgClick, firstTime, actions, children, customAction,
            displayDescription, labels, displayShortName, subName
        } = this.props;
        const {showDropDown} = this.state;
        const queryParam = `name=${encodeURIComponent(breadcrumbName || name)}&back=${encodeURIComponent(window.location.pathname)}`;
        const nonChevronActions = (actions || []).filter(action => action.buttonType !== ButtonType.Chevron);
        const chevronActions = (actions || []).filter(action => action.buttonType === ButtonType.Chevron);
        const showChevronAction = (history && auditLogPath) || firstTime || chevronActions.length > 0;
        return (
            <div className="unit-header-container">
                <div className="unit-header">
                    <div className={`image ${obj.style || ""}`}>
                        {obj.logo && <Logo src={obj.logo}/>}
                        {obj.svg && <obj.svg onClick={() => svgClick && svgClick()}/>}
                        {obj.icon && <FontAwesomeIcon icon={obj.icon}/>}
                    </div>
                    <div className="obj-name">
                        <div className="meta-info-container">
                            <div className="meta-info">
                                {obj.name && <h1>{obj.name}</h1>}
                                {subName && <h6>{subName}</h6>}
                                {obj.organisation &&
                                    <span className="name">
                                        {`${obj.organisation.name}${isEmpty(obj.units) ? "" : " â€“ " + splitListSemantically(obj.units.map(unit => unit.name), I18n.t("service.compliancySeparator"))}`}
                                    </span>}
                                {!isEmpty(labels) &&
                                    <div className="labels">
                                        {labels.sort().map((label, index) => <span key={index}
                                                                                   className="chip-container">
                                        <Chip label={label}/></span>)}
                                    </div>}
                            </div>

                        </div>
                        {(obj.short_name && displayShortName) &&
                            <p className="short-name">{I18n.t("collaboration.shortName")}<span>{obj.short_name}</span>
                            </p>
                        }
                        {(obj.description && displayDescription) &&
                            <MoreLessText txt={obj.description}/>
                        }
                        {children}
                    </div>
                    {!isEmpty(actions) &&
                        <div className="action-menu-container">
                            {nonChevronActions.map((action, index) =>
                                <Button key={index}
                                        onClick={() => !action.disabled && action.perform()}
                                        txt={action.name}
                                        cancelButton={action.buttonType === ButtonType.Secondary}/>)
                            }
                            {showChevronAction &&
                                <div tabIndex={1}
                                     onBlur={() => setTimeout(() => this.setState({showDropDown: false}), 250)}>
                                    <MenuButton txt={I18n.t("home.otherOptions")}
                                                isOpen={showDropDown}
                                                toggle={() => this.setState({showDropDown: !showDropDown})}
                                                buttonType={ButtonType.Secondary}
                                                children={this.otherOptions(chevronActions, firstTime, auditLogPath, history, queryParam)}/>
                                </div>}
                        </div>}
                    {customAction && customAction}
                </div>
            </div>
        )
    }

}

UnitHeader.propTypes = {
    obj: PropTypes.object,
    history: PropTypes.any,
    auditLogPath: PropTypes.string,
    name: PropTypes.string,
    subName: PropTypes.string,
    breadcrumbName: PropTypes.string,
    svgClick: PropTypes.func,
    firstTime: PropTypes.func,
    actions: PropTypes.array,
};

export default UnitHeader;
