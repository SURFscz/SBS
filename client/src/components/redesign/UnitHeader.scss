@import "../../stylesheets/vars";
@import "../../stylesheets/unit_header";

.unit-header-container {
    background-color: $background;
    padding: 15px 0 25px 0;

    @media (max-width: $tablet-max) {
        padding: 15px 15px 25px 15px;
    }

    width: 100%;

    .sds--metadata-list {
        margin-top: auto;
    }

    p.short-name span {
        font-weight: 600;
    }


    .icon-list-items {
        margin-top: 15px;
        .icon-list-item {
            display: flex;
            align-items: center;
            gap: 10px;
            margin-top: 10px;
        }

        span {
            &.not-connected, &.not-available {
                color: var(--sds--color--red--400)
            }

            &.all-connected {
                font-weight: 600;
            }

        }

        svg {
            width: 24px;
            height: auto;
        }
    }

    .unit-header {
        max-width: $medium;
        margin: 0 auto;
        width: 100%;
        display: flex;

        @media (max-width: $compact-medium) {
            flex-direction: column;
        }


        .image {
            display: flex;
            align-items: center;

            svg {
                width: 50px;
                height: 30px;
                color: $primary-orange;
                fill: $primary-orange;

                &.child {
                    color: black;
                    fill: black;
                }
            }

            img {
                max-height: 300px;
                max-width: 300px;
                border-radius: 5px;
            }

            @media (max-width: $compact-medium) {
                display: none;
            }

        }

        .obj-name {
            margin-left: 26px;
            display: flex;
            flex-direction: column;

            .meta-info-container {
                display: flex;

            }

            span.name {
                color: black;
                font-weight: bold;
            }

            span.description {
                margin: 20px 0;
            }
        }

        .labels {
            margin: 10px auto 0 0;
            padding-right: 55px;

            span.chip-container {
                display: inline-block;
                margin: 0 10px 10px 0;
            }
        }

    }

    .action-menu-container {
        display: flex;
        flex-direction: column;
        gap: 20px;
        margin-left: auto;

        button {
            width: 100%;
            min-width: 150px;

            &.sds--btn--secondary {
                min-width: 180px;
            }
        }
    }

    .other-options, .sds--user-info--dropdown {
        background-color: $background;
    }

    .sds--menu-button .sds--user-info--dropdown {
        z-index: 99;
    }

}
