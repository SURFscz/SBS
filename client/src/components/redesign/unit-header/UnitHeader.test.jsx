import React from "react";
import { render } from "@testing-library/react";
import UnitHeader from "./UnitHeader";

describe("UnitHeader", () => {
    it("renders with default props", () => {
        const { asFragment } = render(
            <UnitHeader
                unit={{ name: "Test Unit" }}
                user={{ admin: true }}
                onEditUnit={() => {}}
                onDeleteUnit={() => {}}
                obj={{style: "test-style"}}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
