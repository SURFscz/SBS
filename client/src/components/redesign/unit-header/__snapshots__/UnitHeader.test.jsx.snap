// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitHeader renders with default props 1`] = `
<DocumentFragment>
  <div
    class="unit-header-container"
  >
    <div
      class="unit-header"
    >
      <div
        class="image test-style"
      />
      <div
        class="obj-name"
      >
        <div
          class="meta-info-container"
        >
          <div
            class="meta-info"
          />
        </div>
      </div>
      <div
        class="action-menu-container"
      />
    </div>
  </div>
</DocumentFragment>
`;
